### PR TITLE
Dev rajesh

### PR DESCRIPTION
**Pull request by saini9485**
Main Section Design added

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
-import { RightSideBar } from "./components/RightSideBar/RightSideBar";
+import 'antd/dist/antd.css';
```
Comment (Souvikdeb2612) [inline src/App.js]: Css imports should be done last

File: src/components/Crypto_Main_Chart/MainChart.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  LabelList
+} from "recharts";
+
+const data = [
```
Comment (Souvikdeb2612) [inline src/components/Crypto_Main_Chart/MainChart.jsx]: Create a separate folder for data and create this data there. Finally import it here to use

File: src/components/Crypto_Main_Chart/MainChart.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  },
+  {
+    name: "July",
+    uv: 3490,
+    pv: 4300,
+    amt: 2100
+  }
+];
+
+const CustomizedLabel = (props) => {
```
Comment (Souvikdeb2612) [inline src/components/Crypto_Main_Chart/MainChart.jsx]: directly destructure props here

File: src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import "./TopBuyTopSell.css"
+
+export  function TopBuyTopSell() {
+  return (
+    <div className='top_buySell_container'>
+    <div className='current_balance'>Current Investment
+    <h2>$45,665</h2></div>
```
Comment (Souvikdeb2612) [inline src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx]: Will these values change after integrating the API?

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
+            output_size: 'compact'
+          },
+   headers: {
+    'X-RapidAPI-Key': '82a76f8ad3mshe05dc70539423d5p18ba88jsndb26b70a6c41',
+    'X-RapidAPI-Host': 'alpha-vantage.p.rapidapi.com'
+   }
+        })
+      .then((response) => {
+
+          console.log(response.data[Object.keys(response.data)[1]][Object.keys(response.data[Object.keys(response.data)[1]])[0]]["1. open"]);
```
Comment (Souvikdeb2612) [inline src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx]: remove console statements before raising pr

File: src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx [hunk 6]
```diff
@@ -1,14 +1,52 @@
...
+          setPost([...response.data["Time Series (5min)"]])
+        
+      });
+  }, []);
+console.log({post})
```
Comment (Souvikdeb2612) [inline src/components/Crypto_TopBuy_TopSell/TopBuyTopSell.jsx]: same as above

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (saini9485) [reply src/App.js]: done sir

File: src/components/Crypto_Main_Chart/MainChart.jsx (hunk 3 above)
Comment (saini9485) [reply src/components/Crypto_Main_Chart/MainChart.jsx]: done sir

File: src/components/CryptoAll_Coins/AllCoins.jsx [hunk 7]
```diff
@@ -1,12 +1,68 @@
...
+      });
+  }, []);
+
   return (
-    <div className="top_coins_container">
-      <div className="Bitcoin coin_box">BTC/USD</div>
-      <div className="Ethereum coin_box">ETH/USD</div>
-      <div className="DogeCoin coin_box">DEG/USD</div>
+    <div className="top_coin_card">
+      {topCoin.map((coin, i) => {
```
Comment (Souvikdeb2612) [inline src/components/CryptoAll_Coins/AllCoins.jsx]: destructure coin here

File: src/components/CryptoAll_Coins/AllCoins.jsx [hunk 8]
```diff
@@ -1,12 +1,68 @@
...
+        if (i < 3) {
+          return (
+            <>
```
Comment (Souvikdeb2612) [inline src/components/CryptoAll_Coins/AllCoins.jsx]: make a separate component for all the returned code and call the component here

File: src/components/ExploreCoins/Crypto.jsx [hunk 9]
```diff
@@ -54,6 +54,7 @@ export function Crypto() {
         </div>
         {filteredCoins.map((coin) => {
```
Comment (Souvikdeb2612) [inline src/components/ExploreCoins/Crypto.jsx]: destructure coin

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/ExploreCoins/Crypto.jsx (hunk 9 above)
Comment (saini9485) [reply src/components/ExploreCoins/Crypto.jsx]: done sir

File: src/components/CryptoAll_Coins/AllCoins.jsx (hunk 7 above)
Comment (saini9485) [reply src/components/CryptoAll_Coins/AllCoins.jsx]: done sir

Comment (Souvikdeb2612) [reply src/components/CryptoAll_Coins/AllCoins.jsx]: It is not done

File: src/components/CryptoAll_Coins/AllCoins.jsx (hunk 8 above)
Comment (Souvikdeb2612) [reply src/components/CryptoAll_Coins/AllCoins.jsx]: This is also not done

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/components/CryptoAll_Coins/AllCoins.jsx (hunk 8 above)
Comment (saini9485) [reply src/components/CryptoAll_Coins/AllCoins.jsx]: sir pls check it is right or not

Comment (saini9485) [reply src/components/CryptoAll_Coins/AllCoins.jsx]: sorry sir i did't push the code now i pushed

File: src/routes/Router.jsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { Routes, Route, Navigate } from "react-router-dom";
+import {Crypto} from "../components/exploreCoins/Crypto"
+import Blog from "../pages/Blog"
```
Comment (Souvikdeb2612) [inline src/routes/Router.jsx:3]: format imports

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**